util/journaldb/src/overlayrecentdb.rs

```diff
@@ -195,7 +195,13 @@ impl OverlayRecentDB {
 				};
 				while let Some(rlp_data) = db.get(col, &encode(&db_key)).expect("Low-level database error.") {
 					trace!("read_overlay: era={}, index={}", era, db_key.index);
-					let value = decode::<DatabaseValue>(&rlp_data);
+					let value = match decode::<DatabaseValue>(&rlp_data) {
+						Ok(v) => v,
+						Err(e) => {
+							trace!("read_overlay: Error decoding DatabaseValue err={}, era={}, index{}", e, era, db_key.index);
+							continue;
+						}
+					};
 					count += value.inserts.len();
```

Comment:
What's the reasoning for continuing the loop in this case, instead of the `expect` like in the other places? Couldn't figure it out from a quick read of the code.

Reply:
@ascjones that's a good q but I don't have a great answer. I let it `continue` thinking that the previous code would have done the same but now that you point it out I think it would have panicked. Furthermore the code as I've written it is probably not going to yield a valid&usable `JournalOverlay` if some parts are missing. I'll change it to panic.

Reply:
Yeah at least that way it behaves exactly how it would have before (panic with invalid rlp)